New Booker Component (preparations for booker atom)

JeroenReumkens opened the pull request:
## Adding our new booker component 

![CleanShot 2023-03-22 at 09 34 51](https://user-images.githubusercontent.com/2969573/226845651-1ee5b3eb-3826-49fb-81b4-3b195da74227.gif)

### Summary
The PR adds the new booker component that later will also be used as a separate Atom, i.e. a published package. 
In this PR I've created a new booker component, which purpose is having a component that is totally isolated from SSR data. This way we can later bundle this component as a package and have it fully "self sustainable" in functionality — it will fetch all its data via API's. 

### Component interface
The booker only needs a user slug and an event slug, in order to work. In the below screenshot you see 2 optional props you can add (`rescheduleBooking` and `isAway`). The booking is an SSR provided prop, adding the current booking that is being rescheduled. This is done via SSR since we don't want to expose a public API to fetch meeting data. This also mean that you can not reschedule with only the booker component, if you don't have any server access to provide the booking's data — in other words; rescheduling should always happen on a Cal.com platform. 
The `isAway` boolean gives us an option to show this message early, instead of needing to do an API call for this and see a flash of a loading screen, and after that disappointing the user. 

![CleanShot 2023-03-22 at 09 28 54@2x](https://user-images.githubusercontent.com/2969573/226844058-92e56e45-d2d8-4642-802c-1c365dc945b2.jpg)

### Layout / design
The new booker has 3 layout toggles. In the GIF above you also see them briefly. However, this PR mainly focusses on having the functionality of the 3 layouts, rather than also finishing these 3 different layouts. Sean wanted to focus on the last 2 layouts. Meaning that as soon as we merge this we will disable the buttons at first, and release the new layouts one by one when they're finished. Especially the multi day view is already 90% finished, so could be enabled pretty soon. 
 
![CleanShot 2023-03-22 at 09 36 12@2x](https://user-images.githubusercontent.com/2969573/226845881-14589b9c-188c-4127-beb5-27f42a6af54b.jpg)


## When visiting the test env you need to to the following first:
Below steps need to be followed in order to enable the new booker. Currently both the current + new booker run side by side, this way we can release and test, without breaking existing stuff, and even offer people the option to test the booker themselves, simply by calling this api endpoint via a button. 
1. Go to the latest deployment url 
2. Visit `api/newbooker/enable` to set a cookie that enables you to view the new (temporary) routes
3. From that point on all booker pages will show the new booker page (on the same url) by rewriting the url to contain the `/new-booker/` path, thus pointing to the new pages.

## Videos explaining more about the changes
I advise you to watch any of the below videos (I think the Around summarises it the best), to get a better grasp of what's changed and why. The videos explain more about the changes, giving a voice over to the code, etc. 

### 30 min video going through the code with Alex

https://around.co/playback/e13343f2-8353-40c2-9da5-b770abe0264d?sharedKey=d623a847-db1f-4479-b012-afe5f988fa6d

### Shorter loom videos

https://www.loom.com/share/1584764c43094fad9e2f1088448d8d18

https://www.loom.com/share/fe9285acd9a2465bac928385992b302a

https://www.loom.com/share/cc945c17ce86476ca85758bab41ae27b

https://www.loom.com/share/72800978b1a84f6bbc59ca10b165784f

## Open todo's
- [x] Manual testing: 
- [x] General, 
- [x] Embed (will remain old booker for now, because we can't use cookies in embed)
- [x] Team view
- [x] Dynamic view
- [x] Telemetry?
- [x] Adding initial data via query params
- [x] Eth signature gate
- [x] Metadata 
- [x] Write automated tests
- [x] Small styling nits
- [x] Check if there's new changes in old booker pages that need to be incorporate
- [x] Fix form validation errors — they throw wrong keys now with new form builder, still need to fix that. 
- [ ] Before merge disable toggle for big calendar / 5 day view, since those styles aren't finished fully (and large calendar isn't in there at all). This is work that Sean wanted to finish <3 

## Feedback zomars
- [x] Make invisible redirect instead of /new-booker path (Worked on by Alex)

## Feedback Carina: 
- [x] Reschedule directly shows form instead of first selecting time slot
- [x] Don't show datepicker on mobile
- [x] Align back button and submit button of form on bottom of form

## Feedback Hariom:
- [x] Appearance isn't used
- [x] Powered by cal missing
- [x] Remove timezone select when user is in booking form state
- [x] All booked today message seems to be missing
- [x] Booker doesn't support embed properly yet (known issue, will be discussed down below — decided to only move embed to new booker once the "old" booker isn't used anymore.)

Testing round 2
- [x] Broken timezone select after main merge
- [x] Reschedule doesn't add booking details
- [x] User avatars missing
- [x] Booking error yellow should be blue
- [x] Console warning where classname is added to fragment 

## Feedback Alex
- [x] Duration always added as query param, while not always needed
- [x] Team events without hosts crashes  

## Fixes
#5798
#5768
#5745
#5742
#5763
#6698


<!-- Please provide a loom video for visual changes to speed up reviews
 Loom Video: https://www.loom.com/
-->

**Environment**: Staging(main branch)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

- [ ] Book general event 
- [ ] Book general event with multiple durations
- [ ] Book general event with payment
- [ ] Book general event with multiple occurrences 
- [ ] Book team event
- [ ] Book dynamic event 
- [ ] Validate that pricing formatting is correct, because we now use `Intl.NumberFormat()`instead of large `react-intl` method in `packages/features/bookings/components/event-meta/Price.tsx`
- [ ] Ensure old booker is also still working (little code involved with the old booker is touched, but we still need to test it because a few small things **are** touched.)

JeroenReumkens committed:
Wip on booker atom

JeroenReumkens committed:
Wip on booker atom

JeroenReumkens committed:
Added correct icon imports

sean-brydon commented on packages/atoms/booker/types.ts:
```diff
@@ -0,0 +1,55 @@
+}
+
+/**
+ * @TODO: Abstract this into a reusable type? Or even create a context/zustand
+ * people can wrap so they only have to set this once? Although I doubt that
+ * people will wrap multiple components in one page and thus one context.
```
Smart - we can do this context or zustand - i personally vote zustand as its a little lighter and will allow us to move this out of react if we ever choose to too. zustand stores don't depend on react

JeroenReumkens committed:
Fixed build

JeroenReumkens committed:
Responsive improvements

JeroenReumkens committed:
Removed package lock

JeroenReumkens committed:
Responsive tweaks

JeroenReumkens committed:
Animation improvements and cleanup

JeroenReumkens committed:
Animation improvements and event meta layout improvements.

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Tweaked margins.

JeroenReumkens committed:
Added more event meta blocks

JeroenReumkens committed:
Layout tweaks

JeroenReumkens committed:
Converted booker layout to css grid and implemented multiple layout options

JeroenReumkens committed:
cleanup

JeroenReumkens committed:
Fixed build

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Fixed build

JeroenReumkens committed:
Added temporary api route to enable/disable new booker

JeroenReumkens committed:
Added sticky behavior

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Reverted yarn.lock and reinstalled new packages to see if this fixes build on vercel.

JeroenReumkens committed:
Ensure divider lines always have 100% height.

JeroenReumkens committed:
Improved animation config + initial load

JeroenReumkens committed:
Ensure to pass eventid to getschedule, otherwise custom availability schedule wont work and wont return any availability

JeroenReumkens committed:
Fixed divider line heights in booker

JeroenReumkens committed:
Fixed timezone select positioning

JeroenReumkens committed:
Added ability to view multiple days of timeslots

JeroenReumkens committed:
Added icons to booker toggle

JeroenReumkens committed:
Always show timeslots in timeslots view, also if no date is selected yet. In that case we show upcoming 5 days.

JeroenReumkens committed:
Fixed timeslots in small calendar view

JeroenReumkens committed:
Show selected day in calendar

JeroenReumkens committed:
Fixed booker timeslots view

JeroenReumkens committed:
Wip in making booking form work

JeroenReumkens committed:
Moved most of the booker atom stuff to features, since it belongs there. Atom should be a rather small wrapper.

JeroenReumkens committed:
Added create event functionality to booker form.

JeroenReumkens committed:
Added guests toggle to booker form and styled input addons in dark mode.

JeroenReumkens committed:
Added dynamic weekstart to booker

JeroenReumkens committed:
Added seats limit feature to timeslots.

JeroenReumkens committed:
Removed todo

JeroenReumkens committed:
Added correct event avatars

JeroenReumkens committed:
Added correct event name and icons

JeroenReumkens committed:
Added correct translation for minutes text in multi duration

JeroenReumkens committed:
Add rescheduling functionality to new booker.

JeroenReumkens committed:
Added selected booking time to booking meta in sidebar.

JeroenReumkens committed:
Abstracted away timeformat to custom hook

JeroenReumkens committed:
Added correct key props to all components in booker.

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Fix build

JeroenReumkens committed:
Create some new custom hooks to have a lot less repitition in code.

JeroenReumkens committed:
Moved bookerform component inside booker directory since it is tied to it.

JeroenReumkens committed:
Added error messages to booker form, plus fixed bug in recurring events.

JeroenReumkens commented on apps/web/components/booking/pages/BookingPage.tsx:
```diff
@@ -587,7 +586,8 @@ const BookingPage = ({
                     <div className="text-sm font-medium">
-                      {(rescheduleUid || !eventType.recurringEvent?.freq) && `${parseDate(date, i18n)}`}
+                      {(rescheduleUid || !eventType.recurringEvent?.freq) &&
+                        `${parseDate(date, i18n.language)}`}
```
I made the `parseDate` function a bit easier to use by only passing in the language instead of the full i18n object. This function is moved to the features directory further down since I also had to use it in the new booker.

JeroenReumkens commented on apps/web/components/error/error-page.tsx:
```diff
@@ -1,6 +1,6 @@
 import React from "react";
 
-import { HttpError } from "@lib/core/http/error";
+import { HttpError } from "@calcom/lib/http-error";
```
Moved more stuff to a shared package.

JeroenReumkens commented on apps/web/pages/[user]/book.tsx:
```diff
@@ -3,6 +3,8 @@ import type { GetServerSidePropsContext } from "next";
 import type { LocationObject } from "@calcom/app-store/locations";
 import { privacyFilteredLocations } from "@calcom/app-store/locations";
 import { getAppFromSlug } from "@calcom/app-store/utils";
+import getBooking from "@calcom/features/bookings/lib/get-booking";
```
More shared stuff. (And from here on I'll stop commenting this. If you see such movements, it's because the booker also needed this, and the functionality seemed like a shared thing to me)

JeroenReumkens commented on packages/atoms/booker/Booker.tsx:
```diff
@@ -0,0 +1,12 @@
+import { Booker as BookerComponent, BookerProps } from "@calcom/features/bookings/Booker";
```
We decided that this PR should first be about making a new booker component, later (in a new PR) we'll be turning this into a reuable package. So anything in `packages/atoms` can be ignored for now, it's a veeery rough and incomplete basis. Initial challenges I could already think about are summed up in todo's in this file.

JeroenReumkens commented on packages/features/bookings/Booker/Booker.tsx:
```diff
@@ -0,0 +1,193 @@
+        )}>
+        <AnimatePresence>
+          <StickyOnDesktop key="meta">
+            <BookerSection area="meta">
```
The `area` prop of the `BookerSection` component helps you place the component anywhere in the grid. If you look at the timeslots component around line 165 you also see you can pass a default and a specific grid area for a layout. This way the components can move around in grid position, based on their layout.

JeroenReumkens commented on packages/features/bookings/Booker/Booker.tsx:
```diff
@@ -0,0 +1,193 @@
+};
+
+export const Booker = (props: BookerProps) => (
+  <LazyMotion features={domAnimation}>
```
Lazy loads framer motion for smaller bundle size

JeroenReumkens commented on packages/features/bookings/Booker/components/BookEventForm/CustomInputFields.tsx:
```diff
@@ -0,0 +1,169 @@
+import { EventTypeCustomInputType } from "@prisma/client";
```
Existing custom input fields logic, only abstracted it to it's own reusable file.

JeroenReumkens commented on packages/features/bookings/Booker/components/BookEventForm/GuestFields.tsx:
```diff
@@ -0,0 +1,70 @@
+import type { UseFormReturn } from "react-hook-form";
```
Existing html, abstracted into separate file for readability.

JeroenReumkens commented on packages/features/bookings/Booker/components/DatePicker.tsx:
```diff
@@ -0,0 +1,37 @@
+import { shallow } from "zustand/shallow";
```
Tiny wrapper around existing datepicker component. This components provides all data necessary for the booker, this way colocates everything involved in the booker datepicker into one isolated file for readability.

JeroenReumkens commented on packages/features/bookings/Booker/components/EventMeta.tsx:
```diff
@@ -0,0 +1,57 @@
+import { m } from "framer-motion";
```
Left part of the booker — contains all "meta" info of an event.

JeroenReumkens commented on packages/features/bookings/Booker/components/EventMeta.tsx:
```diff
@@ -0,0 +1,57 @@
+          <EventMembers schedulingType={event.schedulingType} users={event.users} profile={event.profile} />
+          <EventTitle className="mt-2 mb-8">{event?.title}</EventTitle>
+          <div className="space-y-5">
+            <EventDetails event={event} />
```
Reusable component with all event details, abstracted away outside the booker directory for reusability. See that component for more details.

JeroenReumkens commented on packages/features/bookings/Booker/components/EventMeta.tsx:
```diff
@@ -0,0 +1,57 @@
+              </span>
+              <TimezoneSelect
+                menuPosition="fixed"
+                className="[&_.cal-react-select\_\_control]:h-auto [&_.cal-react-select\_\_control]:min-h-0 [&_.cal-react-select\_\_control]:cursor-pointer [&_.cal-react-select\_\_control]:border-0 [&_.cal-react-select\_\_control]:ring-0 [&_.cal-react-select\_\_indicators]:hidden [&_.cal-react-select\_\_menu]:w-[300px] [&_.cal-react-select\_\_menu-portal]:z-30"
```
Don't like this styling, but we can improve it once the old booking page is gone and we can touch the timezoneselect component.

JeroenReumkens commented on packages/features/bookings/Booker/components/LargeCalendar.tsx:
```diff
@@ -0,0 +1,26 @@
+import dayjs, { Dayjs } from "@calcom/dayjs";
```
This is a placeholder for Sean's amazing component <3

JeroenReumkens commented on packages/features/bookings/Booker/utils/event.ts:
```diff
@@ -0,0 +1,36 @@
+import { useTimePreferences } from "../../lib/timePreferences";
+import { useBookerStore } from "../store";
+
+export const useEvent = () => {
```
Small hook that abstracts away the trpc query, and right away passed in the correct data from the booker store. (this file is located inside booker directory, so only supposed to be used in there).

JeroenReumkens commented on packages/features/bookings/components/event-meta/Details.tsx:
```diff
@@ -0,0 +1,167 @@
+  EventDetailBlocks.PRICE,
+];
+
+export const EventMetaBlock = ({
```
Event meta block is reusable component that can be used to show event information — it takes in an icon and children, mainly ensuring the layout is correct. It's also used in the EventDetails component down below.

JeroenReumkens commented on packages/features/bookings/lib/timePreferences.ts:
```diff
@@ -0,0 +1,34 @@
+import { create } from "zustand";
```
Tiny hook and store that makes the time settings super accessible and easy to use <3

JeroenReumkens commented on packages/lib/parse-dates.ts:
```diff
@@ -1,23 +1,22 @@
+import { parseZone } from "./parse-zone";
 
-const processDate = (date: string | null | Dayjs, i18n: I18n) => {
+const processDate = (date: string | null | Dayjs, language: string) => {
```
Existing code move from apps/web to reusable function. Only changed i18n argument to make it easier to use. Also updated implementation in web.

JeroenReumkens commented on packages/trpc/server/routers/viewer.tsx:
```diff
@@ -142,6 +146,112 @@ const publicViewerRouter = router({
     }),
   // REVIEW: This router is part of both the public and private viewer router?
   slots: slotsRouter,
+  event: publicProcedure
```
New public query to fetch event data. We really have to double check this to ensure we're not leaking any data here that shouldn't be publicly queryable.

JeroenReumkens commented on packages/ui/components/badge/Badge.tsx:
```diff
@@ -20,14 +21,24 @@ const classNameBySize = {
-} & JSX.IntrinsicElements["div"];
+};
+
+export type BadgeProps =
```
Updated badge component to also be a button instead of a clickable div. This button-badge is better for a11y.

JeroenReumkens commented on packages/ui/components/form/inputs/Input.tsx:
```diff
@@ -12,18 +12,23 @@ import { FiEye, FiEyeOff, FiX } from "../../icon";
 import { HintsOrErrors } from "./HintOrErrors";
 import { Label } from "./Label";
 
```
Added dark mode styles to input fields. This way I could remove a lot of custom styles from the booker form components 🫶

JeroenReumkens commented on apps/web/lib/mutations/bookings/create-booking.ts:
```diff
@@ -1,10 +0,0 @@
-import * as fetch from "@lib/core/http/fetch-wrapper";
```
It's not deleted by moved to `packages/features/bookings/lib/create-booking.ts`.

JeroenReumkens committed:
Added some comments <3

JeroenReumkens committed:
Fixed todos in booker form.

JeroenReumkens committed:
Added loading state for timeslot selector, and added prefetching of next month, in case of multi day view showing 2 months at the same time.

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Fixed import paths

JeroenReumkens committed:
Added away view

JeroenReumkens committed:
Validate uniqueness of event attendees.

JeroenReumkens commented on apps/web/pages/api/newbooker/[status].tsx:
```diff
@@ -0,0 +1,24 @@
+import type { NextApiRequest, NextApiResponse } from "next";
```
Temporary api route to enabled/disable new url.

JeroenReumkens commented on packages/atoms/booker/Booker.tsx:
```diff
@@ -0,0 +1,13 @@
+import type { BookerProps } from "@calcom/features/bookings/Booker";
```
Any file in `packages/atom` is preparation for the usage of this component as atom. However, to speed up the implementation on our side we decided to first implement the new booker in our own website, before finding solutions for things like dynamic urls, translations, importing next's Link component which we don't want in the atom/package (because that bundles a lot of next as dependency), etc etc. So feel free to ignore this `atoms` directory for now, quite some work left still after the booker itself is merged.

JeroenReumkens commented on packages/features/bookings/Booker/Booker.tsx:
```diff
@@ -0,0 +1,196 @@
+import { Fragment, useEffect } from "react";
+import StickyBox from "react-sticky-box";
+import { shallow } from "zustand/shallow";
+
```
This file is the main entrypoint for the booker. This loads all different sections.

JeroenReumkens commented on packages/features/bookings/Booker/Booker.tsx:
```diff
@@ -0,0 +1,196 @@
+    shallow
+  );
+
+  useInitializeBookerStore({
```
By storing the most important data of the booker in a Zustand store, we have easy access to all this data in child components, without prop drilling.

JeroenReumkens commented on packages/features/bookings/Booker/Booker.tsx:
```diff
@@ -0,0 +1,196 @@
+    <>
+      {/* @TODO: This is imported because the DatePicker relies on it. Perhaps
+      rewrite the datepicker as well? */}
+      <CustomBranding lightVal={null} darkVal={null} />
```
I didn't want to touch or duplicate the datepicker. So for now that means we need to load the custombranding component as well. As soon as we delete the old booker I want to improve the datepicker + this custombranding component as well.

JeroenReumkens commented on packages/features/bookings/Booker/Booker.tsx:
```diff
@@ -0,0 +1,196 @@
+};
+
+export const Booker = (props: BookerProps) => {
+  if (props.isAway) return <Away />;
```
This away state is passed into the booker component. This can be either done client side, but in our own case we already fetch this boolean server side, so we can show the away state asap without running api calls client side.

JeroenReumkens commented on packages/features/bookings/Booker/components/Away.tsx:
```diff
@@ -0,0 +1,22 @@
+import { useLocale } from "@calcom/lib/hooks/useLocale";
```
Couldn't find a design for this, so for now I took the in current version. Will check with Ciaran.

JeroenReumkens commented on packages/features/bookings/Booker/store.ts:
```diff
@@ -0,0 +1,154 @@
+import { useEffect } from "react";
+import { create } from "zustand";
+
```
Booker store that contains all shared booker data and data manipulation methods that are used by it's child components.  See interface comment for more details.

JeroenReumkens committed:
Tweaked comment

JeroenReumkens commented on packages/features/bookings/lib/create-recurring-booking.ts:
```diff
@@ -1,17 +1,10 @@
-};
-
-const createRecurringBooking = async (data: ExtendedBookingCreateBody[]) => {
+// @TODO: Didn't look at the contents of this function in order to not break old booking page.
```
Moved to reusable place, contents isn't touched, things have been made a bit easier to share though.

JeroenReumkens committed:
#5798 added correct date format and style for selected date in booker.

JeroenReumkens committed:
UI improvements

JeroenReumkens committed:
Enable possibility to add booking values via query params.

JeroenReumkens committed:
Added functionality to update query params when user selects date/duration etc in booker

JeroenReumkens committed:
First steps in adding e2e test.

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Fixes after merge with main, and added new form builder.

JeroenReumkens committed:
Implemented new form types and validation to booker, confirming new form builder. Validation still throwing wrong error keys though.

JeroenReumkens committed:
Added search to timezone dropdown

JeroenReumkens committed:
Added e2e test for booker (copy of current booker tests, only enabling cookie), plus fixed reschedule view.

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Updated yarn.lock

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Added new booker for team pages.

JeroenReumkens committed:
Fixed input addon (hover) styles.

JeroenReumkens committed:
Added dynamic booking.

JeroenReumkens committed:
Hide timeformat  select for multi day view for now.

JeroenReumkens commented on apps/web/pages/[user].tsx:
```diff
@@ -180,7 +182,9 @@ export default function User(props: inferSSRProps<typeof getServerSideProps> & E
                     prefetch={false}
                     href={{
-                      pathname: `/${user.username}/${type.slug}`,
+                      pathname: newBookerEnabled
```
Temporary. This way if you have the cookie, you directly go to the new page without having to change the url manually.

JeroenReumkens commented on apps/web/playwright/new-booking-pages.e2e.ts:
```diff
@@ -0,0 +1,134 @@
+test.describe("free user", () => {
+  test.beforeEach(async ({ page, users, context }) => {
+    const free = await users.create();
+    context.addCookies([{ name: "new-booker-enabled", value: "true", url: "http://localhost:3000" }]);
```
Test is the same test as the current booking page test, except that I need to enable the cookie. Apart from that all tests were green after adding the right test ids 🥳

JeroenReumkens commented on packages/features/bookings/Booker/components/BookEventForm/BookEventForm.tsx:
```diff
@@ -0,0 +1,267 @@
+import { zodResolver } from "@hookform/resolvers/zod";
```
Abstracted away the form in its own component as well. This way we can test it separately instead of as part of the bigger page. We could decide in the future to make this fully reusable, but for now it relies on the booker's store to get its data, instead of having to pass everything in as props.

JeroenReumkens commented on packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx:
```diff
@@ -0,0 +1,122 @@
+import { useFormContext } from "react-hook-form";
```
New work by hariom copied straight from there.

JeroenReumkens committed:
Cleanup and ui tweaks

JeroenReumkens committed:
removed log

JeroenReumkens committed:
Mobile improvements

JeroenReumkens committed:
Cleanup

JeroenReumkens committed:
Small design tweaks after talking to ciaran.

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Text color and weight tweaks in booker

JeroenReumkens committed:
Added rainbow gates to new booker.

JeroenReumkens committed:
Added in default values which fixes form vallidation (???).

JeroenReumkens committed:
Added empty defaults for name and email

JeroenReumkens committed:
Added metadata

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Reset yarn.lock

JeroenReumkens committed:
Fixed booker zod validation after change in main.

JeroenReumkens committed:
Icon tweak

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Fixed timezone select styles after new classnames have been merged.

JeroenReumkens committed:
Updated seat availability styles.

zomars committed:
Merge branch 'main' into feat/new-booker-atom

zomars committed:
Update yarn.lock

zomars reviewed (commented):
Partial review. Will continue tomorrow.

JeroenReumkens commented on .env.example:
```diff
@@ -177,3 +177,5 @@ CSP_POLICY=
 # Vercel Edge Config
 EDGE_CONFIG=
+
+ALCHEMY_API_KEY=
```
Very good point. I needed this in order to test the web3 (rainbow) implementation / gate. Will add an explanation to the best of my abilities  😁 Might be worth double checking it.

PeerRich replied:
let me double check but we may wanna sunset our web3 app due to low usage

PeerRich replied:
but we'll do that in a follow up PR

JeroenReumkens committed:
Added explanation for alchemy key to .env.example

JeroenReumkens commented on packages/features/eventtypes/lib/getPublicEvent.ts:
```diff
@@ -0,0 +1,149 @@
+    recurringEvent: isRecurringEvent(event.recurringEvent) ? parseRecurringEvent(event.recurringEvent) : null,
+    // Sets user data on profile object for easier access
+    profile: {
+      username: event.users[0].username,
```
Interesting. I can't seem to replicate this. Of course I can add a check on whether an event has users, but I'm more curious in which cases that can happen. Can you explain a bit more about how you could trigger this?

hariombalhara replied:
I think it can happen for a team event which has no users added.

hariombalhara replied:
Also, you may want to use `owner` property as first preference.

zomars replied:
I was in a collective event with 4 users added

JeroenReumkens committed:
Added tooltip to booker month/week/multiday toggle

JeroenReumkens committed:
Fixed timezoneselect styles in booker after select updates.

PeerRich commented on apps/web/components/Gates.tsx:
```diff
@@ -48,4 +53,38 @@ const Gates: React.FC<GateProps> = ({ children, gates, appData, dispatch }) => {
+ * (instead of accepting it as props) and then passing it along
+ * to the Gates component.
+ */
+export const BookerGates = ({ children }: { children: React.ReactNode }) => {
+  const event = useEvent();
+  const setEthSignature = useBookerStore((state) => state.setEthSignature);
+  const rainbowAppData = event?.data ? getEventTypeAppData(event.data, "rainbow") || {} : {};
+  const gates = [];
+  if (rainbowAppData && rainbowAppData.blockchainId && rainbowAppData.smartContractAddress) {
+    gates.push("rainbow");
+  }
+
+  const [gateState, gateDispatcher] = useReducer(
+    (state: GateState, newState: Partial<GateState>) => ({
+      ...state,
+      ...newState,
+    }),
+    {}
+  );
+
+  useEffect(() => {
+    setEthSignature(gateState.rainbowToken);
+  }, [gateState, setEthSignature]);
+
+  return (
+    <Gates gates={gates} appData={rainbowAppData} dispatch={gateDispatcher}>
+      {children}
+    </Gates>
+  );
+};
```
does this always run, even if rainbow is disabled?

zomars replied:
Also we shouldn't load extra JavaScript for booking pages that don't need it

JeroenReumkens committed:
Updates bookingfields component by taking changes from current booker component

JeroenReumkens committed:
Removed remaining booker todos

JeroenReumkens committed:
Fix bookeventform

JeroenReumkens committed:
Fix for recurring event meta

zomars commented:
Solving a conflict with @Udit-takkar recent PR #7642

zomars committed:
Merge branch 'main' into feat/new-booker-atom

zomars commented:
Solved conflicts

zomars committed:
Type fixes

zomars committed:
Typefixes

zomars committed:
Team event fixes

zomars reviewed (commented):
- Fixes some typing issues
- Fixed collective team bookings errors
- Found a weird error on the calendar, possibly related with hydration mismatch: https://www.loom.com/share/a50f6e351cd54f0984e908f334cf5524

zomars commented on apps/web/middleware.ts:
```diff
@@ -32,6 +32,12 @@ const middleware: NextMiddleware = async (req) => {
   }
 
+  // Allow visiting new booker page when cookie is present.
+  if (url.pathname.includes("/new-booker/") && !req.cookies.has("new-booker-enabled")) {
```
Can we make invisible rewrite so people with the cookie can see the new booker without going to `"/new-booker/"`?

JeroenReumkens replied:
@zomars  Great idea! Not sure how we would implement that though, because:

1. In middleware we would need to match ALL routes instead of only the paths in `config.matcher`, which would run a lot more middleware invocations? Plus not sure how we can detect if we're actually looking at a booking page because it only contains out of 2 dynamic parts? 
2. We also can't run it in geterversideprops, because `/[user]/[type]` is a statically generated page. 

Would love to hear how you would solve this.

zomars commented on packages/features/bookings/types.ts:
```diff
@@ -0,0 +1,33 @@
+};
+
+export type BookingResponse = Awaited<
+  ReturnType<typeof import("@calcom/features/bookings/lib/handleNewBooking").default>
```
Updated to better match the actual response

zomars commented on packages/features/eventtypes/lib/getPublicEvent.ts:
```diff
@@ -0,0 +1,193 @@
+  userMetadata as userMetadataSchema,
+} from "@calcom/prisma/zod-utils";
+
+const publicEventSelect = Prisma.validator<Prisma.EventTypeSelect>()({
```
Updated to have valid types on select

zomars commented on packages/features/eventtypes/lib/getPublicEvent.ts:
```diff
@@ -0,0 +1,193 @@
+    bookingFields: getBookingFieldsWithSystemFields(event),
+    recurringEvent: isRecurringEvent(event.recurringEvent) ? parseRecurringEvent(event.recurringEvent) : null,
+    // Sets user data on profile object for easier access
+    profile: getProfileFromEvent(event),
+    users: getUsersFromEvent(event),
```
Added to handle team events

JeroenReumkens commented:
> * Found a weird error on the calendar, possibly related with hydration mismatch: https://www.loom.com/share/a50f6e351cd54f0984e908f334cf5524

Thanks for all the fixes @zomars  This day shifting isn't something I noticed 🤯 I will dive into this today, also see if I can solve the hydration error, because that's annoying in development as well.

CarinaWolli commented:
Great work 👏🏻   Here are some issues I found when testing: 

- when clicking on reschedule I am redirected to the final booking page without being able to pick a slot first:
![Screenshot 2023-03-27 at 13 11 53](https://user-images.githubusercontent.com/30310907/227925638-80348587-7714-4760-b2d1-352223b01885.png)

- on mobile view we shouldn't show the date picker anymore once a slot is chosen: 
![Screenshot 2023-03-27 at 11 40 07](https://user-images.githubusercontent.com/30310907/227926126-bb3bce51-2387-41e5-b0ba-8da4381dbd22.png)

- NIT: Confirm and back button is not on the bottom when there are no locations or additional fields:
![Screenshot 2023-03-27 at 10 55 29](https://user-images.githubusercontent.com/30310907/227926194-3baeb44d-cbb4-49cf-a253-af4923e7e96c.png)

JeroenReumkens committed:
Avoid hydration errors by only rendering date picker client side. Remove web3 gates since we dont offer them anymore. Prevent timeslot select from staying open when switching to a different month.

JeroenReumkens committed:
Don't show calendar on mobile booker during booking.

JeroenReumkens committed:
Always align booker buttons to bottom

JeroenReumkens commented:
> Great work 👏🏻 Here are some issues I found when testing:
> 
> * when clicking on reschedule I am redirected to the final booking page without being able to pick a slot first:
>   ![Screenshot 2023-03-27 at 13 11 53](https://user-images.githubusercontent.com/30310907/227925638-80348587-7714-4760-b2d1-352223b01885.png)
> * on mobile view we shouldn't show the date picker anymore once a slot is chosen:
>   ![Screenshot 2023-03-27 at 11 40 07](https://user-images.githubusercontent.com/30310907/227926126-bb3bce51-2387-41e5-b0ba-8da4381dbd22.png)
> * NIT: Confirm and back button is not on the bottom when there are no locations or additional fields:
>   ![Screenshot 2023-03-27 at 10 55 29](https://user-images.githubusercontent.com/30310907/227926194-3baeb44d-cbb4-49cf-a253-af4923e7e96c.png)

@CarinaWolli  Thanks for testing so thoroughly like always 🙏 

1. Works now after another fix I did, we verified together. 
2. Fix pushed
3. Fix pushed

Thanks again!

JeroenReumkens committed:
Don't show backend messages in error, rather show a helpful text like the current booker does as well.

hariombalhara commented:
Testing and leaving comments as and when I see them
First one
- Appearance doesn't seem to have an impact on the booker
<img width="1729" alt="Screenshot 2023-03-27 at 6 13 39 PM" src="https://user-images.githubusercontent.com/1780212/227943775-89a7e90d-0cf4-47bd-8adc-5fee949e1bfc.png">

hariombalhara commented:
Testing and leaving comments as and when I see them
First one
- Appearance doesn't seem to have an impact on the booker
<img width="1729" alt="Screenshot 2023-03-27 at 6 13 39 PM" src="https://user-images.githubusercontent.com/1780212/227943775-89a7e90d-0cf4-47bd-8adc-5fee949e1bfc.png">

emrysal commented:
I've got a solution for invisible rewrites, will PR to here once Github allows pushes again.

I found some minor issues as well:

1) Duration seems to always add to the URL even when it is not needed.

2) On team events, when there is no hosts there's a bit of a crash

<img width="824" alt="image" src="https://user-images.githubusercontent.com/1046695/227940212-84bc702f-4252-4438-aafb-446cef8fb3bb.png">

This used to be:
<img width="848" alt="image" src="https://user-images.githubusercontent.com/1046695/227940339-0f56b536-7b07-4788-886c-0ce260daf838.png">

hariombalhara commented:
<img width="658" alt="Screenshot 2023-03-27 at 8 13 57 PM" src="https://user-images.githubusercontent.com/1780212/227974882-14733641-4f0d-49bb-bad8-1dac0d3cb821.png">
"Powered by Cal.com" is missing. Checked the code, it's not there.

emrysal committed:
Do invisible next rewrite based on cookie from next.config.js (#7949)

* Do invisible next rewrite based on cookie from next.config.js

* Name embed link instead of bookerPath

* Rewrites only dynamic user pages

---------

Co-authored-by: zomars <zomars@me.com>

JeroenReumkens commented:
Thanks all for your comments on the booker @zomars @CarinaWolli @hariombalhara @emrysal. I've added all your points as todo's in the main issue description, and will check them once completed. Will let you know if all of them are completed so you can recheck. 

Some things I want to discuss upfront: 

@hariombalhara 
1. Appearance isn't used is correct. The reason is because @sean-brydon  is currently working on new theming CSS variables, so I wanted to wait to the very last minute before implementing the current solution, as that might be redundant very soon. I've added an unchecked todo in the description though, so we can decide what to do with this before merge. 
2. As for the Embed: Yes you're right, the embed shouldn't use the new booker yet. Only once it's fully released. That's one of the main reasons I left the current booker in. 

@emrysal 
3. What do you mean by "duration is aded __also when not needed__" — you mean when there's only a single duration? 

Thanks again all, I will work on the other feedback and let you know when it's fixed.

JeroenReumkens committed:
Don't allow change of timezone when bookerform is visible

JeroenReumkens committed:
Don't add duration to query param if the event is not a multi duration event.

zomars committed:
Update next.config.js

JeroenReumkens committed:
Added correct timezone formatting to event meta when timeslot is selected.

JeroenReumkens committed:
removed .env variable that isn't needed anymore.

zomars committed:
Update Gates.tsx

zomars committed:
Type fixes

zomars committed:
Allows to run all tests with the new booker

zomars reviewed (commented):
Added the cookie to all e2e tests so we can see which edge cases are failing with the new booker

zomars committed:
Merge branch 'main' into feat/new-booker-atom

zomars commented:
Solved conflicts

hariombalhara commented:
Took the latest pull and started re-testing.
- Broken UI for timezone selector <img width="914" alt="Screenshot 2023-03-28 at 1 12 13 PM" src="https://user-images.githubusercontent.com/1780212/228164372-46e83a9b-8df9-49eb-9576-1d31c6b0aaa3.png">
- Reschedule doesn't show the Booking questions's responses for the original booking. This is visible in the screenshot as well that Carina shared <img width="499" alt="Screenshot 2023-03-28 at 1 16 40 PM" src="https://user-images.githubusercontent.com/1780212/228165438-74c78d72-d704-4668-a05b-878993e5a098.png">
- Seeing just names of users <img width="995" alt="Screenshot 2023-03-28 at 1 21 32 PM" src="https://user-images.githubusercontent.com/1780212/228166718-9b6cae3c-a47e-4247-808e-83fa7b150547.png">
- This is in a bluish tone in production, not sure if it's intentional to convert it into orange <img width="704" alt="Screenshot 2023-03-28 at 1 29 18 PM" src="https://user-images.githubusercontent.com/1780212/228168884-154969b1-0d02-462d-9345-a703b003ab69.png">
- Seeing this warning in console <img width="1039" alt="Screenshot 2023-03-28 at 1 31 03 PM" src="https://user-images.githubusercontent.com/1780212/228169337-6010f459-cf5d-427d-b558-f8d1b9d9aaa3.png">

JeroenReumkens committed:
Fixed timezone select styles after merge.

JeroenReumkens committed:
Don't throw error when event doesn't have hosts, rather return no users, which will result in no availability in UI.

JeroenReumkens committed:
Make booker errors of severity info instead of warning.

JeroenReumkens committed:
Ensure team avatars are shown, as well as filter on uniqueness of avatars.

JeroenReumkens committed:
Added all booked today message to timeslots.

JeroenReumkens committed:
Added cal.com logo to booker.

JeroenReumkens committed:
Fixed fragment classname error, minor mobile animation tweaks plus make all booked today text smaller for multi day layout.

JeroenReumkens committed:
Improved timezone select styles, and updated arguments of getbooking function after updates in main.

JeroenReumkens committed:
Prevent infinite loop in rewriting new booker.

JeroenReumkens committed:
Prevent infinite loop in rewriting new booker.

zomars commented on apps/web/next.config.js:
```diff
@@ -207,7 +207,9 @@ const nextConfig = {
-        source: `/:user((?!${pages.join("|")}).*)/:type`,
+        // Excludes the new-booker pages from the rewrite,
+        // because otherwise this would case an infinite loop.
+        source: `/:user((?!${pages.join("|")}).*)/:type(!new-booker)`,
```
Nice find @JeroenReumkens (regex is hard!) 🙏🏽

JeroenReumkens committed:
Moved new-booker pages to their own directory to prevent regexes confusing next and thus nut running getserversideprops after rewrite. Also adding clearing of old date in booker store, that could stick around when user immediately navigates back to the same page after booking.

JeroenReumkens committed:
Fixed cal logo color in darkmode for new booker.

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Implemented new color tokens and theme variables. Also small design tweaks after merge with main.

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Minor style tweaks

JeroenReumkens committed:
Show multiple locations in tooltip on booker #8222

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Radio button style tweaks

JeroenReumkens committed:
Fixed build

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Updated calendar imports to new lucide names

JeroenReumkens committed:
Removed resetting of selected times logic, because otherwise url params wouldnt be taken into account which is actually what we want. So old values sticking around when navigating back is actually the desired behavior.

JeroenReumkens committed:
Updated tests to instead of always run the new booker in tests, have a utility to run both the new and old booker for specified tests.

JeroenReumkens committed:
Added comment and eslint disable for if statement in booker test.

sean-brydon commented on packages/features/bookings/Booker/Booker.tsx:
```diff
@@ -0,0 +1,213 @@
+          style={resizeAnimationConfig.small_calendar.default as MotionStyle}
+          animate={resizeAnimationConfig[layout]?.[bookerState] || resizeAnimationConfig[layout].default}
+          transition={{ ease: "easeInOut", duration: 0.4 }}
+          className={classNames(
+            "[--booker-meta-width:280px] [--booker-main-width:480px] [--booker-timeslots-width:240px] lg:[--booker-timeslots-width:280px]",
+            "bg-muted grid max-w-full items-start overflow-clip dark:[color-scheme:dark] md:flex-row",
+            layout === "small_calendar" &&
+              "border-subtle mt-20 min-h-[450px] w-[calc(var(--booker-meta-width)+var(--booker-main-width))] rounded-md border",
+            layout !== "small_calendar" && "h-auto min-h-screen w-screen"
+          )}>
```
Im not against all these arbitrary values but as much as they are being used here would it not be easier to do this in style tag?

sean-brydon replied:
Youre not wrong its deffo neater :P

sean-brydon commented on packages/features/bookings/Booker/Booker.tsx:
```diff
@@ -0,0 +1,213 @@
+    rescheduleBooking,
+  });
+
+  useEffect(() => {
+    setLayout(isMobile ? "mobile" : "small_calendar");
+  }, [isMobile, setLayout]);
```
Would this be better using `useLayoutEffect` ideally we just need the dom to have been loaded and not the whole component for this i believe?

JeroenReumkens replied:
Using `useLayoutEffect` would actually do a first render that's visible to the user, and only then update the layout. So in theory that could result in first seeing layout A, and only after seeing the mobile layout? I'm wondering what the benefit of it would be?

sean-brydon replied:
Youre right - think i was miss understanding what it does in my head :P 

Just re-read what it actually does - youre right this is not what we need

sean-brydon commented on packages/features/bookings/Booker/utils/event.ts:
🔥

sean-brydon commented on packages/features/bookings/components/event-meta/Details.tsx:
```diff
@@ -0,0 +1,168 @@
+    <>
+      {blocks.map((block) => {
+        if (typeof block === "function") {
+          return <Fragment key={block.name}>block(event)</Fragment>;
```
```suggestion
          return <Fragment key={block.name}>{block(event)}</Fragment>;
```

JeroenReumkens committed:
Update packages/features/bookings/components/event-meta/Details.tsx

Co-authored-by: sean-brydon <55134778+sean-brydon@users.noreply.github.com>

sean-brydon commented on packages/ui/components/badge/Badge.tsx:
```diff
@@ -41,19 +40,50 @@ type IconOrDot =
+    </>
+  );
+
+  if (isButton) {
+    return (
+      <button {...passThroughProps} className={classes}>
+        <Children />
+      </button>
+    );
+  }
+
+  return (
+    <div {...passThroughProps} className={classes}>
+      <Children />
     </div>
   );
```
We can clean this up a bit 

```suggestion
  const Wrapper  = isButton ? "button" ? "div"

  return (
    <Wrapper {...passThroughProps} className={classes}>
      <Children />
    </Wrapper>
  );
```

sean-brydon replied:
We might get some type issues - not tested this

JeroenReumkens replied:
TypeScript doesn't like that unfortunately 😭 Try it. If you can make it work you're my hero 🫶

sean-brydon replied:
Fixed with this solution :) Pushed fix 

<img width="981" alt="Screenshot 2023-04-17 at 14 57 14" src="https://user-images.githubusercontent.com/55134778/232507611-32923d1e-8148-415a-9d04-49bbe7df6bb7.png">

sean-brydon commented on packages/ui/components/form/timezone-select/TimezoneSelect.tsx:
```diff
@@ -49,13 +50,15 @@ export function TimezoneSelect({
             state.isFocused && "bg-subtle",
             state.isSelected && "bg-emphasis text-default",
-            props.classNames?.option
+            timezoneClassNames?.option && timezoneClassNames.option(state)
```
oops :P

sean-brydon replied:
Note to self - if state is passed in as a function you need to call it

sean-brydon reviewed (commented):
Tested a lot of this - looked through all the code left some suggestions and talking points. 

Awesome job! I really love the use of zustand being brought into this. A lot better than propdrilling and 1000% better than using context :P 

Also a great change if we ever decide to make this component framework agnostic <3 

👏

sean-brydon committed:
Fix badge types

zomars committed:
Merge branch 'main' into feat/new-booker-atom

sean-brydon committed:
Merge branch 'main' into feat/new-booker-atom

PeerRich committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Lazy loaded timezone select to save 85kb in bundle size.

JeroenReumkens committed:
Upgraded framer to latest. Als moved framer and react sticky deps to features instead of atoms.

zomars reviewed (dismissed):
- Unit tests are failing
- E2E tests are failing
- We have a new conflict

I want to get this merged ASAP to prevent more conflicts. But we need to fix those checks 🙏🏽

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Added new pagewrapper logic

JeroenReumkens committed:
Simplified rescheduling ssr fetches, this now also supports multi seat rescheduling.

JeroenReumkens committed:
Unset selected time when user is rescheduling directly after a new booking, otherwise it would show the form instead of new time selection.

JeroenReumkens committed:
Updated form builder logic as per form builder in current booker.

JeroenReumkens committed:
Updated form builder prefill logic as per logic in current booker.

JeroenReumkens committed:
Updated getbooking function to fetch correct details when a reschedule uid is used

PeerRich committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Fixed booking questions test by NOT waiting for /book page because the new booker doesnt have this.

JeroenReumkens committed:
Added former meeting time to reschedule view.

JeroenReumkens committed:
Fixed types

JeroenReumkens committed:
Undo playwright config update by mistake.

JeroenReumkens committed:
Fixed event types test by only waiting for /book page in old booker

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Set new booker cookie to one year in the future instead of 2050

CarinaWolli commented:
Functionality-wise everything worked for me, so nothing blocking from my side 🙌🏻 There are just some more design issues/questions, I feel like some of it wasn't like that when I first tested it 🤔 

**1. Screenshot:** 
![Screenshot 2023-04-20 at 10 00 11](https://user-images.githubusercontent.com/30310907/233310613-f27741ea-de8e-4ece-bb82-1fc9b5eb885a.png)
- Shouldn’t the background be white? I think it is bg-muted right now 
- We don’t scroll anymore when the booking description is too long
- The back and confirm button is still not on the bottom right
- Does the text on the left side have the correct font color and font weight? 
- I feel like we should have a little more space on the left so that we can at least fit the slot date in one line

**2. Screenshot:** 
<img width="749" alt="Screenshot 2023-04-20 at 10 23 54" src="https://user-images.githubusercontent.com/30310907/233311024-8e2af467-22fd-4c55-8d4b-07c315a02e37.png">
- It should show the recurring dates here and not 'every 5 weeks for 12 occurrences'

**3. Screenshot:** 
<img width="312" alt="Screenshot 2023-04-20 at 10 27 40" src="https://user-images.githubusercontent.com/30310907/233311869-3bd7ae86-e28b-4ea7-b614-8a3f1d048605.png">
- Big gap on mobile between booking information and input fields

**4. Screenshot:** 
<img width="262" alt="Screenshot 2023-04-20 at 10 29 53" src="https://user-images.githubusercontent.com/30310907/233311992-13632b08-7f35-43fe-99ba-1961754bc30c.png">
- I think the team avatar used to not be shown here

JeroenReumkens committed:
added reset mockdate to test

PeerRich commented:
the dot indicating the day needs better positioning

![CleanShot 2023-04-20 at 23 45 23@2x](https://user-images.githubusercontent.com/8019099/233494446-2525a1ba-1f0c-4f77-afae-ce6eed6298d3.jpg)

PeerRich commented:
this needs love:

![CleanShot 2023-04-20 at 23 49 18@2x](https://user-images.githubusercontent.com/8019099/233494992-25397f34-46ae-468d-9219-79f2aca6efd9.jpg)

PeerRich commented:
there are viewports that dont work (but used to work). i think because we force some min-width but also overflow hidden this breaks

https://user-images.githubusercontent.com/8019099/233495036-aba535b3-bfe2-41af-bba7-1ff45718d1d3.mp4

PeerRich commented:
this looks too crowded/close to each other: 

![CleanShot 2023-04-20 at 23 50 47@2x](https://user-images.githubusercontent.com/8019099/233495237-ff1ea26b-f258-4893-b7eb-f85d3a59d7fb.jpg)

PeerRich commented:
it would be nice if the animation is also applied when changing routes (animate the container size)

https://user-images.githubusercontent.com/8019099/233495477-c297aaa2-bea2-412a-9861-39787ed47565.mp4

PeerRich commented:
imo the time buttons are too huge and spacing looks odd

![CleanShot 2023-04-20 at 23 53 33@2x](https://user-images.githubusercontent.com/8019099/233495678-5651a76c-56b1-467f-941d-72b2ccc50f98.jpg)

i like the old better:

![CleanShot 2023-04-20 at 23 54 20@2x](https://user-images.githubusercontent.com/8019099/233495771-fc1985a8-f005-4376-95a9-9560456ba77f.jpg)

PeerRich commented:
this looks weird:

![CleanShot 2023-04-20 at 23 59 08@2x](https://user-images.githubusercontent.com/8019099/233496484-fe02ced6-1142-4663-9662-89f5e6376e5f.jpg)


looks better previously: 

![CleanShot 2023-04-20 at 23 59 41@2x](https://user-images.githubusercontent.com/8019099/233496563-df07e54c-9141-4828-a896-526244ce6a7a.jpg)


I think because the buttons are not aligned at the bottom and the height is unusually higher than before

PeerRich commented:
Layout shift when starting to type:
https://user-images.githubusercontent.com/8019099/233497017-0b1d2d03-7463-467b-ab2b-9bc47058b1bd.mp4

doesn't happen in v1

PeerRich committed:
Merge branch 'main' into feat/new-booker-atom

PeerRich commented:
not sure if I like: "All booked today." here

first, it should probably not be "today" and second, I wonder if we should either (a) hide that day if unavailable or (b) have like disabled times there



![CleanShot 2023-04-21 at 00 05 29@2x](https://user-images.githubusercontent.com/8019099/233497320-afb6e627-6c3e-4a6c-b6d5-39eba948a115.jpg)

Jaibles commented:
Anyone else finding issues when reviewing this, please add them to the [Booking Page project in Linear](https://linear.app/calcom/project/booking-page-20-4eb862750686).

JeroenReumkens committed:
Temporary disabled test to see if this solves the out of memory error.

JeroenReumkens committed:
Merge branch 'main' into feat/new-booker-atom

JeroenReumkens committed:
Deleted test to see if that fixes the memory error

PeerRich reviewed (approved):
tested and worked decent enough 

can get merged since its behind a feature flag

future fixes in new follow up PRs

JeroenReumkens committed:
Select first day when switching months in booker

zomars reviewed (approved):
let's do this!

Jaibles commented:
> not sure if I like: "All booked today." here
> 
> first, it should probably not be "today" and second, I wonder if we should either (a) hide that day if unavailable or (b) have like disabled times there or similar. probably (a)
> 
> ![CleanShot 2023-04-21 at 00 05 29@2x](https://user-images.githubusercontent.com/8019099/233497320-afb6e627-6c3e-4a6c-b6d5-39eba948a115.jpg)

Designs have that day hidden if it's unavailable.
